### ISSUE TITLE
snowman

### ISSUE THREAD

**Issue by mrexodia** [Cararasu/holodec#3]
Not sure if you're aware of [snowman](https://github.com/yegord/snowman). Perhaps you could take ideas/x86 semantics from that and not reinvent the wheel too hard...

**Comment by mrgreywater** [comment]
Not to forget [fcd](https://github.com/zneak/fcd). I don't want to discourage anyone, but it would probably make more sense to focus the efforts.

**Comment by Cararasu** [comment]
Yeah, I was aware of snowman but did not look at how it did things. The goal of the whole project was to make a decompiler from scratch and not overly rely on all kinds of frameworks to do the job. I could have used someone else's solution to raise x86 to an IR but I had the desire to develop it my own way ;-).
fcd also seems interesting, but it raises everything into LLVM, which I am not convinced is the best representation to solve the problem. I could be wrong and my approach is worse, but that is something I have to find out as I go along.

**Comment by mrexodia** [comment]
Snowman doesn't use a framework, I'm not saying you should use their code, but all kinds of quirks in x86 are documented and implemented there so it might save some time...

**Comment by Cararasu** [comment]
oh, i misunderstood. Yeah, I had my fair share of problems trying to find out how certain x86 instructions work. The whole architecture is a mess. I am always surprised that processors work at all ;-).
Anyway, I just found it and will take a look at it tomorrow. At the moment I am trying to clean up code I already wrote, so its a good time to maybe get inspirations for improvements.
